cdap-api/src/main/java/co/cask/cdap/internal/api/DefaultDatasetConfigurer.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.api;
+
+import co.cask.cdap.api.DatasetConfigurer;
+import co.cask.cdap.api.data.stream.Stream;
+import co.cask.cdap.api.data.stream.StreamSpecification;
+import co.cask.cdap.api.dataset.Dataset;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.module.DatasetModule;
+import co.cask.cdap.internal.dataset.DatasetCreationSpec;
+import com.google.common.base.Preconditions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of {@link DatasetConfigurer} for adding datasets and streams in Configurers.
+ */
+public class DefaultDatasetConfigurer implements DatasetConfigurer {
```

Comment:
The idea of Configurer interface is that only the interface is in cdap-api so that we are free to implement it inside cdap-app-fabric without worrying about bring in extra library dependencies (which we cannot do with the old configure() method API).